scratch/controller/controller.cpp

```diff
@@ -0,0 +1,127 @@
+#include "../../src/network/network.hpp"
+#include <linux/joystick.h>
+#include <stdlib.h>
+#include <stdio.h>
+#include <iostream>
+#include <fcntl.h> 
+#include <unistd.h>
+
+const unsigned short 
+    //Button values
+    A_BUTTON = 0,
+    B_BUTTON = 1,
+    X_BUTTON = 2,
+    Y_BUTTON = 3,
+    LB_BUTTON = 4,
+    RB_BUTTON = 5,
+    BACK_BUTTON = 6,
+    START_BUTTON = 7,
+    HOME_BUTTON = 9,
+    //Axis values
+    JOYSTICK_LEFT_X = 0,
+    JOYSTICK_LEFT_Y = 1,
+    JOYSTICK_RIGHT_X = 3,
+    JOYSTICK_RIGHT_Y = 4,
+    DPAD_X = 6,
+    DPAD_Y = 7,
+    LT_AXIS =2,
+    RT_AXIS =5;
+//Axis ranges from -32767 to +32767
+//If recalibrate is used, the input will be read between 
+//POSITIVE: MIN_POS and MAX_POS
+//NEGATIVE: -MAX_POS and -MIN_POS
+//and then scaled to fit range -32767 to 32767.
+//Essentially making MIN_POS and less the new 0 or 'dead' value.
+const int16_t 
+    MAX_POS = 32767,
+    MIN_POS = 5000;
+struct movement {
+    int16_t left, right;
+};
+
+//Joystick values range from -32767 to +32767
+//Used formula scaled_val = (new_max - new_min) / (old_max - old_min) * (v - old_min) + new_min
+//Can the rover handle the calibration side instead?
```

Comment:
This is good, I want the calibration on the base station side so it can be changed on the fly if necessary.